src/charm.py

```diff
@@ -749,15 +795,17 @@ def _mongos_layer(self) -> Layer:
     @property
     def mongos_initialised(self) -> bool:
         """Check if mongos is initialised."""
-        return "mongos_initialised" in self.app_peer_data
+        return json.loads(self.app_peer_data.get("mongos_initialised", "false"))
 
     @mongos_initialised.setter
     def mongos_initialised(self, value: bool):
```

Comment:
```suggestion
    def mongos_initialised(self, value: bool=False):
```
I *believe* this change is needed based on what you have below

Reply:
This is a property setter, a default value is useless here because it will never be used.
A property setter can only be called as `self.mongos_initialised = <SOMETHING>`.